detox/src/devices/runtime/drivers/ios/XCUITestUtils.js

```diff
@@ -0,0 +1,223 @@
+    '-allowProvisioningUpdates',
+    '-destination', `platform=iOS Simulator,id=${simulatorId}`,
+    'test-without-building'
+  ];
+
+  const env = {
+    TEST_RUNNER_IS_DETOX_ACTIVE: '1',
+    TEST_RUNNER_DETOX_SERVER: detoxServer,
+    TEST_RUNNER_DETOX_SESSION_ID: detoxSessionId,
+    TEST_RUNNER_TEST_TARGET_SERVER_PORT: testTargetServerPort,
+    TEST_RUNNER_BUNDLE_ID: bundleId,
+    TEST_RUNNER_DETOX_DEBUG_VISIBILITY: debugVisibility,
+    TEST_RUNNER_DETOX_DISABLE_VIEW_HIERARCHY_DUMP: disableDumpViewHierarchy
+  };
+
+  const options = {
+    maxBuffer: 1024 * 1024 * 10, // 10MB
+  };
+
+  const command = `${env.TEST_RUNNER_IS_DETOX_ACTIVE ? Object.keys(env).map(key => `${key}=${env[key]}`).join(' ') : ''} ${xcodebuildBinary} ${flags.map(flag => flag.includes(' ') ? `"${flag}"` : flag).join(' ')}`;
+  log.debug(`[XCUITest] Running command: ${command}`);
+
+  const isRunningOnTerminal = process.stdout.isTTY;
+  if (isRunningOnTerminal !== true) {
+    log.debug(`[XCUITest] Currently not running through the Terminal, trying to execute the command from the Terminal`);
+
+    try {
+      return _runCommandInTerminal(command, options);
+    } catch (e) {
+      log.warn(`[XCUITest] Failed to execute the command from the Terminal, falling back to the default execution`);
+    }
+  }
+
+  const result = exec(command, options);
+
+  result.childProcess.stderr.on('data', (data) => {
+    if (data.includes('Testing started completed')) {
+      result.childProcess.kill('SIGTERM');
+    }
+  });
+
+  return result;
+}
+
+function _runCommandInTerminal(command, options) {
+  const escapedCommand = command.replace(/"/g, '\\"');
+
+  // Opens the Terminal app and runs the command in a new window, then closes the window when the command is done running.
+  const appleScript = `
+        tell application "Terminal"
+            if not running then
+                open
+            end if
+
+            activate
+            do script "${escapedCommand}; exit"
+
+            repeat
+                delay 1
+                if not busy of window 1 then
+                    close window 1
+                    exit repeat
+                end if
+            end repeat
+        end tell
+    `;
+
+  return exec(`osascript -e '${appleScript}'`, options);
+}
+
+function _allowNetworkPermissionsXCUITest() {
+  log.debug(`[XCUITest] Allowing network permissions`);
+  let didCallback = false;
+
+  const childProcess = osascript.executeFile(
+    `${__dirname}/allowNetworkPermissionsXCUITest.scpt`,
+    function(err, _, __) {
+      if (err) {
+        log.error(`[XCUITest] Failed to approve network permissions for XCUITest target:\n\t${err}`);
+      } else {
+        log.debug(`[XCUITest] Network permissions are allowed`);
+      }
+
+      didCallback = true;
+    });
+
+  setTimeout(() => {
+    if (didCallback || !childProcess) {
+      return;
+    }
+
+    log.debug(`[XCUITest] Killing the process that allows network permissions (timed-out)`);
```

Comment:
Well , usually you can pass a timeout and killSignal to a process to make code simpler:

![image](https://github.com/wix/Detox/assets/1962469/716b3f3a-0f86-406e-a991-e9402a31483c)